packages/core/src/components/spinner/spinner.tsx

```diff
@@ -81,16 +81,18 @@ export class Spinner extends AbstractPureComponent<ISpinnerProps, {}> {
 
         return (
             <TagName className={classes}>
-                <svg height={size} width={size} viewBox="0 0 100 100" strokeWidth={strokeWidth}>
-                    <path className={Classes.SPINNER_TRACK} d={SPINNER_TRACK} />
-                    <path
-                        className={Classes.SPINNER_HEAD}
-                        d={SPINNER_TRACK}
-                        pathLength={PATH_LENGTH}
-                        strokeDasharray={`${PATH_LENGTH} ${PATH_LENGTH}`}
-                        strokeDashoffset={strokeOffset}
-                    />
-                </svg>
+                <span className={Classes.SPINNER_ANIMATION}>
```

Comment:
you have previously lamented the lack of comments in this part of the code, which caused many commits and reversions. So, add a whole bunch of comments here as to why we need so many wrappers.

Reply:
https://github.com/palantir/blueprint/pull/2890/files#diff-85bf4db926db7eec3120880625d59adbR42

Reply:
comments in styles

Reply:
and now here too 👍